### DISCUSSION TITLE
Distilling Knowledge from a image classification model with sigmoid function and binary cross entropy

### DISCUSSION THREAD

**Discussion by publioelon** [yoshitomo-matsubara/torchdistill#213] [Q&A]
Hello, I'd like to ask if it's possible to use this framework to use a custom trained model (input shape 128x160) and two classes fever and healthy to a smaller model, given that it used binary cross-entropy for loss computation and sigmoid function for classification? I need to compress the model size and switch to softmax classification due to layer compatibility with hardware accelerator ([Edge TPU google coral accelerator](https://coral.ai/products/accelerator/)).

**Comment by yoshitomo-matsubara** [comment]
Hi @publioelon 

Let me summarize your setting:
1. You use a custom dataset, which has only 2 classes
2. Input shape is 128x160
3. Your teacher model is trained with 1. and 2.
(Correct me if I am wrong ^)

To apply conventional KD in PyTorch, you would use output from the final linear layer without softmax/sigmoid as CrossEntropyLoss and KLDivLoss are designed to use them.
If you are not sure how to do that, share your model code here so that I can tell you where you should make the minor change.

Besides that, which Dataset class in PyTorch are you using for your custom dataset?
Once you provide these two (model and dataset) details, I can tell you the exact steps
